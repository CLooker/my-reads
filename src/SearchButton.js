import React, { Component } from 'react';
import SearchBar from './SearchBar';
import './App.css';

const SearchButton = function(props) {
	return (
		<div className="open-search">
      <a onClick={props.addBook}>Add a book</a>
    </div>
  )
}

export default SearchButton;