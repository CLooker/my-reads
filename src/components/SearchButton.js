import React from 'react';
import { Link } from 'react-router-dom';

const SearchButton = ({ resetSearch, ...location }) => (
  <div className="open-search">
    <Link onClick={resetSearch} to="my-reads/search">
      Add a book
    </Link>
  </div>
);

export default SearchButton;
