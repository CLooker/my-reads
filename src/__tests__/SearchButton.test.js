import React from 'react';
import { shallow } from 'enzyme';
import SearchButton from '../components/SearchButton';

describe('SearchButton', () => {
  it('should render correctly', () => {
    const componentProps = {
      match: {
        path: 'my-reads'
      }
    };

    const component = shallow(<SearchButton {...componentProps} />);

    expect(component).toMatchSnapshot();
  });
});
