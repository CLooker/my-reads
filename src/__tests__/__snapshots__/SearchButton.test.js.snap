// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render correctly 1`] = `
<div
  className="open-search"
>
  <Link
    replace={false}
    to="my-reads/search"
  >
    Add a book
  </Link>
</div>
`;
